js/components/Drives.tsx

```diff
@@ -32,9 +59,54 @@ export const Drives = ({ io, sectors }: DrivesProps) => {
         number: 2,
         name: 'Disk 2',
     });
+    const [smartData1, setSmartData1] = useState<DiskIIData>({
+        on: false,
+        number: 1,
+        name: 'HD 1'
+    });
+    const [smartData2, setSmartData2] = useState<DiskIIData>({
+        on: false,
+        number: 2,
+        name: 'HD 2'
+    });
+
+    const [smartPort, setSmartPort] = useState<SmartPort>();
+
+    const hash = useHash();
+
+    useEffect(() => {
+        if (smartPort && disk2) {
+            const hashParts = getHashParts(hash);
+            let loading = 0;
+            for (const drive of [1, 2] as DriveNumber[]) {
+                if (hashParts && hashParts[drive]) {
+                    const hashPart = decodeURIComponent(hashParts[drive]);
+                    if (hashPart.match(/^https?:/)) {
+                        loading++;
+                        loadHttpUnknownFile(disk2, smartPort, drive, hashPart, onProgress)
+                            .catch((e) => setError(e))
+                            .finally(() => {
+                                if (--loading === 0) {
+                                    ready.onReady();
+                                }
+                                setCurrent(0);
+                                setTotal(0);
+                            });
```

Comment:
I can't see how the progress bar is going to work. From this, it looks like _all_ of the files will be loaded in parallel, meaning that they will each try to update the progress bar asynchronously. Since they all will be emitting different `current` and `total` values, the progress bar could jerk all over the place if all of the images are coming in over a slow network connection.

The easiest solution is just to wait for the previous to finish before starting the next one. Otherwise, `onProgress` would have to aggregate everything.

Reply:
Yeah, it's a little simplistic, which is how it was before, basically if you load two large files the progress will flicker back and forth. That's not a common use case for now so it generally works. I think it would be easy enough to attach an id with each file and either display multiple bars or aggregate the values.